### ISSUE TITLE
Number of actions

### ISSUE THREAD

**Issue by WangChen100** [zuoxingdong/VIN_TensorFlow#1]
hi, Dong. I am very appreciate your work on VIN. 
**But I wonder why the number of Q layer, ch_q, is set as 10.**
In my view, there are 9 potential positions for each position in the map according to the agent's action.  So it means that there should be 9 actions and 9 layers of Q.

**Comment by zuoxingdong** [comment]
Hi @WangChen100 thank you for your feedback. The motivation here is that each Q layer corresponds to a 'virtual' action, so the number of layers is not necessarily to be the number of valid actions. Empirically, a bit more layers lead to better performance.

**Comment by WangChen100** [comment]
Thank you @zuoxingdong  very much for your reply.  It's a really useful suggestion and trick. And I have another question why activations of convolution in many VIN projects are set **None** as well as yours.
In my mind, if the activation of tf.layers.conv2d is  set None, the two linear convolution layers can be combined into one convolution layer. Waiting for your reply, thank you.

`    
h = tf.layers.conv2d(inputs=X, 
                         filters=ch_h, 
                         kernel_size=[3, 3], 
                         strides=[1, 1], 
                         padding='same', 
                         activation=None,
                         use_bias=True, 
                         kernel_initializer=tf.truncated_normal_initializer(stddev=0.01), 
                         bias_initializer=tf.zeros_initializer(), 
                         name='h0',
                         reuse=tf.AUTO_REUSE)
    r = tf.layers.conv2d(inputs=h, 
                         filters=1, 
                         kernel_size=[3, 3],  
                         strides=[1, 1], 
                         padding='same', 
                         activation=None,
                         use_bias=False,
                         kernel_initializer=tf.truncated_normal_initializer(stddev=0.01), 
                         bias_initializer=None,
                         name='r',
                         reuse=None)
`

**Comment by fanyuzeng** [comment]
More convolution layers increase the parameters of networks.
